posts/views.py

```diff
@@ -428,6 +437,120 @@ def get(self, request, authorid):
                 contentTypes.append( "<p>" + post.content +"</p>")
         return render(request, 'author/author_posts.html', context={'author': author, 'posts':serializer.data, 'contentTypes':contentTypes})
 
+class GetAuthorPosts(views.APIView):
+    authorHelper = FrontEndAuthorPosts()
+
+    def get(self, request, authorid):
+        paginator = CustomPagination()
+        author = self.authorHelper.get_user(authorid)
+        user = request.user
+        friendship_level = self.authorHelper.get_friendship_level(request, author)
+        posts = self.authorHelper.get_feed(user, author, friendship_level)
+        result_page = paginator.paginate_queryset(posts, request)
+        serializer = PostSerializer(result_page, many=True)
+        return paginator.get_paginated_response(serializer.data, "posts")
+
+class FrontEndAuthorPosts(TemplateView):
+    def get_posts(self,author):
+        return Post.objects.filter(author = author)
+
+    def get_user(self, pk):
+        try:
+            return User.objects.get(pk=pk)
+        except User.DoesNotExist:
+            raise Http404
+
+    def get_feed(self,user,other,f_level):
+        # f_level is a list of either ['FOAF'] or ['FOAF','FRIENDS'] or []
+        allPosts = self.get_posts(other)
+        feedPosts = allPosts.filter(visibility__in=['PUBLIC','SERVERONLY'])
+        allPosts = allPosts.exclude(visibility__in=['PUBLIC','SERVERONLY'])
+        for level in f_level:
+            feedPosts = feedPosts.union(allPosts.filter(visibility=level))
+        allPosts = allPosts.filter(visibility='PRIVATE')
+        return feedPosts
+        
+        # add all posts with acceptable friendship
+
+    def get_friendship_level(self,request,other):
+        if(are_friends(request.user,other)):
+            return ['FRIENDS', 'FOAF']
+        if(are_FOAF(request.user,other)):
+            return ['FOAF']
+        return []
+
+    def get(self, request, authorid):
+        # get the allowedVisibilitys of the relationship between request.user ~ authorid
+        author = self.get_user(authorid)
+        user = request.user
+        friendship_level = self.get_friendship_level(request,author)
+        posts = self.get_feed(user,author, friendship_level)
+        serializer = PostSerializer(posts, many=True)
+        contentTypes = []
+        for post in posts:
+            if post.contentType == "text/markdown":
+                contentTypes.append(commonmark.commonmark(post.content))
+            else:
+                contentTypes.append( "<p>" + post.content +"</p>")
+        return render(request, 'author/author_posts.html', context={'author': author, 'posts':serializer.data, 'contentTypes':contentTypes})
+
+class GetAuthorPosts(views.APIView):
```

Comment:
I think this should be moved to the feed view file just because it has to do with the author feeds.